MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -621,40 +1041,77 @@ int main(int argc, char * argv[])
                 {
+                    errorCode = msixmgr::GetErrorCodeFromHRESULT(E_INVALIDARG);
+                    errorDesc = L"Path of the VHD(X) image to be unmounted not given. HRESULT " + errorCode + L".";
+
                     std::wcout << std::endl;
                     std::wcout << "Please provide the path to the image you would like to unmount." << std::endl;
                     std::wcout << std::endl;
+
+                    // Telemetry : Workflow Log
+                    QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                    workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                    msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), false, workflowElapsedTime, errorCode.c_str(), errorDesc.c_str());
+
                     return E_INVALIDARG;
                 }
 
                 HRESULT hrUnmountVHD = MsixCoreLib::UnmountVHD(cli.GetMountImagePath());
 
                 if (FAILED(hrUnmountVHD))
                 {
+                    errorCode = msixmgr::GetErrorCodeFromHRESULT(hrUnmountVHD);
+                    errorDesc = L"Failed Unmounting the VHD(X) file. HRESULT " + errorCode + L".";
+
                     std::wcout << std::endl;
                     std::wcout << "Unmounting the VHD " << cli.GetMountImagePath() << " failed with HRESULT 0x" << std::hex << hrUnmountVHD << std::endl;
                     std::wcout << std::endl;
+
+                    // Telemetry : Workflow Log
+                    QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                    workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                    msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), false, workflowElapsedTime, errorCode.c_str(), errorDesc.c_str());
+
                     return hrUnmountVHD;
                 }
                 else
                 {
                     std::wcout << std::endl;
                     std::wcout << "Successfully unmounted the VHD " << cli.GetMountImagePath() << std::endl;
                     std::wcout << std::endl;
+
+                    // Telemetry : Workflow Log
+                    QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                    workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                    msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), true, workflowElapsedTime, L"", L"");
                 }
             }
             else
             {
+                errorCode = msixmgr::GetErrorCodeFromHRESULT(ERROR_NOT_SUPPORTED);
+                errorDesc = L"Invalid FileType given for Unmount Operation. HRESULT " + errorCode + L".";
+

```

Comment:
Same as above

Reply:
As discussed, all the Telemetry variables will have the same Error Message as the actual Console statements. The change for outputting the Error Message on Console from a variable will be done as part of Error Message Improvement Task that will be taken later.